py/htm/examples/mnist.py

```diff
@@ -67,32 +66,36 @@ def load_images(file_name):
             assert(len(raw) == data_start + img_size * num_imgs)   # All data should be used.
         return imgs
 
-    train_labels = load_labels(os.path.join(path, 'train-labels-idx1-ubyte.gz'))
-    train_images = load_images(os.path.join(path, 'train-images-idx3-ubyte.gz'))
-    test_labels  = load_labels(os.path.join(path, 't10k-labels-idx1-ubyte.gz'))
-    test_images  = load_images(os.path.join(path, 't10k-images-idx3-ubyte.gz'))
+    train_labels = load_labels(os.path.join(path, 'train-labels-idx1-ubyte'))
+    train_images = load_images(os.path.join(path, 'train-images-idx3-ubyte'))
+    test_labels  = load_labels(os.path.join(path, 't10k-labels-idx1-ubyte'))
+    test_images  = load_images(os.path.join(path, 't10k-images-idx3-ubyte'))
 
     return train_labels, train_images, test_labels, test_images
 
-
+# these parameters can be improved using parameter optimization, 
+# see py/htm/optimization/ae.py
+# For more explanation of relations between the parameters, see 
+# src/examples/mnist/MNIST_CPP.cpp 
 default_parameters = {
-    'boostStrength': 7.80643753517375,
-    'columnDimensions': (35415,1),
-    'dutyCyclePeriod': 1321,
-    'localAreaDensity': 0.05361688506086096,
-    'minPctOverlapDutyCycle': 0.0016316043362658,
-    'potentialPct': 0.06799785776775163,
-    'stimulusThreshold': 8,
-    'synPermActiveInc': 0.01455789388651146,
-    'synPermConnected': 0.021649964738697944,
-    'synPermInactiveDec': 0.006442691852205935
+    'potentialRadius': 7,
+    'boostStrength': 7.0,
+    'columnDimensions': (28*28*8, 1),
+    'dutyCyclePeriod': 1402,
+    'localAreaDensity': 0.1,
+    'minPctOverlapDutyCycle': 0.2,
+    'potentialPct': 0.1,
```

Comment:
The SP is making  very few connections, because of the low potential percent and the low potential radius.

> Potential Synapses on Segment Min/Mean/Max 12 / 19.8571 / 23
> Connected Synapses on Segment Min/Mean/Max 6 / 15.0831 / 23

This isn't a problem per say, and it's probably why it runs so fast.

Reply:
> isn't a problem per say, and it's probably why it runs so fast. 

that, and the lower number of columns needed. The riddance is in the cpp experiment (I think we have much more columns than needed, so we can force a column to become more specialized)